tests/unit/single_table/test_base.py

```diff
@@ -104,3 +108,584 @@ def test_fit(self):
+        data = pd.DataFrame({
+            'name': ['John', 'Doe', 'John Doe'],
+            'salary': [90.0, 100.0, 80.0]
+        })
+        instance = Mock()
+        instance._data_processor._dtypes = pd.Series()
+        instance._filter_conditions.return_value = data[data.name == 'John Doe']
+        conditions = {'salary': 80.}
+        transformed_conditions = {'salary.value': 80.0}
+
+        # Run
+        sampled, num_valid = BaseSynthesizer._sample_rows(
+            instance,
+            3,
+            conditions=conditions,
+            transformed_conditions=transformed_conditions
+        )
+
+        # Assert
+        assert num_valid == 1
+        pd.testing.assert_frame_equal(sampled, data[data.name == 'John Doe'])
+        instance._sample.assert_called_once_with(3, {'salary.value': 80.0})
+        instance._data_processor.reverse_transform.assert_called_once_with(
+            instance._sample.return_value
+        )
+        instance._data_processor.filter_valid.assert_called_once_with(
+            instance._data_processor.reverse_transform.return_value
+        )
+
+    def test__sample_rows_with_previous_rows(self):
+        """Test that when calling sample rows with previous rows those are being concatenated."""
+        # Setup
+        previous_rows = pd.DataFrame({
+            'name': ['John Doe', 'John Doe', 'John Doe'],
+            'salary': [80.0, 80.0, 80.0]
+        })
+        data = pd.DataFrame({
+            'name': ['John', 'Doe', 'John Doe'],
+            'salary': [90.0, 100.0, 80.0]
+        })
+
+        instance = Mock()
+        instance._data_processor._dtypes = pd.Series()
+        instance._data_processor.filter_valid = lambda x: x
+        instance._data_processor.reverse_transform.return_value = data
+
+        # Run
+        sampled, num_valid = BaseSynthesizer._sample_rows(instance, 3, previous_rows=previous_rows)
+
+        # Assert
+        expected_data = pd.DataFrame({
+            'name': ['John Doe', 'John Doe', 'John Doe', 'John', 'Doe', 'John Doe'],
+            'salary': [80.0, 80.0, 80.0, 90.0, 100.0, 80.0]
+        })
+        assert num_valid == 6
+        pd.testing.assert_frame_equal(sampled, expected_data)
+        instance._sample.assert_called_once_with(3)
+        instance._data_processor.reverse_transform.assert_called_once_with(
+            instance._sample.return_value
+        )
+
+    def test__sample_rows_type_error(self):
+        """Test when ``_sample`` does not accept ``transformed_conditions``."""
+        # Setup
+        data = pd.DataFrame({
+            'name': ['John', 'Doe', 'John Doe'],
+            'salary': [90.0, 100.0, 80.0]
+        })
+        instance = Mock()
+        instance._data_processor._dtypes = pd.Series()
+        instance._filter_conditions.return_value = data[data.name == 'John Doe']
+        conditions = {'salary': 80.}
+        transformed_conditions = {'salary.value': 80.0}
+        instance._sample.side_effect = [TypeError, None]
```

Comment:
this should be changed to a `NotImplementedError`, or the code in `CTGAN` and `TVAE` needs to change to raise a `TypeError`